Add remove endpoint

JuliaOlshevskaia opened the pull request:
Кроме добавления эндпойнтов для удаления пользователя и фильма, добавила сортировку жанра при создании фильма, иначе не проходили тесты

JuliaOlshevskaia committed:
feat: add recommendation films for user

JuliaOlshevskaia committed:
feat: delete '=='

JuliaOlshevskaia committed:
fix: delete 'drop table'

goloviu committed:
Фича "Лента событий"

feat: добавить ленту событий
test: добавить тесты для новых методов

goloviu committed:
Фича "Лента событий"

feat: добавить ленту событий
test: добавить тесты для новых методов

goloviu committed:
fix: удалить неиспользуемые импорты

goloviu committed:
fix: удалить неиспользуемые импорты

goloviu committed:
fix: удалить неиспользуемые импорты

JuliaOlshevskaia committed:
feat: add recommendation films for user

JuliaOlshevskaia committed:
fix: rename userLikes, otherUserLikes

JuliaOlshevskaia committed:
fix: add checking exist user

JuliaOlshevskaia committed:
fix: add sorted genres for addFilm

anbananova commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmDbStorage.java:
```diff
@@ -54,6 +54,9 @@ public Film remove(Film film) {
 
     @Override
     public Film update(Film film) {
+        String sqlDeleteGenres = "delete from movie_genre where movie_id = ?";
```
чтобы сохранить единый стиль кода, можно сделать основные операторы в запросе заглавными:
```DELETE FROM movie_genre WHERE movie_id = ?```

anbananova reviewed (approved):
все отлично, в одном месте можно подправить стиль кода

JuliaOlshevskaia committed:
fix: style

Yarikov-MY commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmStorage.java:
```diff
@@ -31,4 +31,6 @@ public interface FilmStorage {
 
     boolean isFilmExist(Integer filmId);
+
+    void remove(Integer filmId);
```
Можно добавить тесты для новых методов в storages, позволит проверять функционал после слияний

Yarikov-MY reviewed (changes requested):
Хороший код, можно добавить тесты

goloviu committed:
fix: Заменить параметры метода saveUserFeed с получения eventId, operationId на enum
fix: Добавить плагин в мавен для декодинга в UTF-8 при maven test
fix: добавить CASCADE в sql скрипте

goloviu committed:
Фича "Лента событий"

fix: добавить пробел

goloviu committed:
Фича "Лента событий"

fix: убрать пробел

JuliaOlshevskaia committed:
feat: add tests for delete film and user by id

JuliaOlshevskaia committed:
feat: add tests for delete film and user by id
